packages/api/resolvers/mutations/createProductBundleItem.js

```diff
@@ -1,16 +1,30 @@
 import { log } from 'meteor/unchained:core-logger';
-import { Products } from 'meteor/unchained:core-products';
-import { ProductNotFoundError, InvalidIdError } from '../../errors';
+import { Products, ProductTypes } from 'meteor/unchained:core-products';
+
+import {
+  ProductNotFoundError,
+  InvalidIdError,
+  ProductWrongTypeError,
+} from '../../errors';
 
 export default function createProductBundleItem(root, { productId, item }) {
   log(`mutation createProductBundleItem ${productId}`, { item });
+
   if (!productId) throw new InvalidIdError({ productId });
   if (!item.productId)
     throw new InvalidIdError({ bundleItemId: item.productId });
   const product = Products.findOne(productId);
+
+  const bundleItem = Products.findOne(item.productId);
+  if (!bundleItem)
+    throw new ProductNotFoundError({ productId: item.productId });
   if (!product) throw new ProductNotFoundError({ productId });
-  const bundleProduct = Products.findOne(item.productId);
-  if (!bundleProduct) throw new ProductNotFoundError({ productId });
+  if (bundleItem.type !== ProductTypes.BundleProduct)
```

Comment:
Also this, I think productId product has to be a BundleProduct where as the bundleItem is part of the bundle therefore can be anything.